CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,186 @@
+
+The contract has the following configurable parameters:
+
+1. **`address public approver;`** - Set by Governance. Intended to be a multisig and has the authority to approve exchanges. The signers for the multisig are at the discretion of Governance, but it would comprise of community members or members on behalf of organizations that are aligned with the Celo network/ecosystem.
+2. **`uint256 public exchangeWaitPeriodSeconds;`** - Set by Governance. The minimum amount of time in seconds that must elapse between a proposed exchange being approved and when the exchange can be executed. Should give sufficient time for Governance to veto an approved exchange.
+3. **`FixidityLib.Fraction public spread;`** - Set by Governance. The percent fee imposed upon an exchange execution.
+4. **`mapping(address => ExchangeLimits) stableTokenExchangeLimits`** - Set by Governance via `setStableTokenExchangeLimits`. A mapping of stable token address to a struct containing two `uint256`s, `minExchangeAmount` and `maxExchangeAmount`.
+
+The contract has the following functions:
+
+1. **`function proposeExchange(address stableToken, uint256 sellAmount, bool sellCelo) external returns (uint256)`** - Called by an exchange proposer to propose an exchange.
+   * Callable by anyone.
+   * If `sellCelo` is true, CELO is the asset being sold and `stableToken` is the asset being bought. If `sellCelo` is false, `stableToken` is the asset being sold and CELO is the asset being bought.
+   * Requires the amount of `stableToken` being bought/sold to be within the token-specific range `[stableTokenExchangeLimits[stableToken].minExchangeAmount, stableTokenExchangeLimits[stableToken].maxExchangeAmount]` that is set by Governance via `setStableTokenExchangeLimits` (described later).
+   * Deposits the full amount of the asset being sold into the contract.
+   * Records:
+     * The exchange as in the Proposed state.
+     * A struct with the following info is stored in a mapping with an `id` key:
+       * The stable token address.
+       * The amount of the asset being sold.
+       * Whether CELO is being sold.
+   * Returns:
+     * The `id` of the struct in the mapping.
+2. **`function approveExchangeProposal(uint256 id) external`** - Approves a proposed exchange.
+   * Only callable by the approver address.
+   * Marks the proposed exchange with the given `id` as approved.
+   * Records:
+     * The exchange as in the Approved state.
+     * The timestamp at which the approval occurred (ie `block.timestamp`).
```

Comment:
Maybe just blockNumber so it's easier to look up.

Reply:
Not sure I follow why blockNumber is more advantageous?

Reply:
you can just paste it on the explorer and it's easier to get the approve tx and tx that happened in the same block that could be relevant. This is definitely a nit though.

Reply:
ahhh this isn't intended to be for looking up the approval tx, this records the timestamp so there can be a check when executing that the appropriate amount of time has elapsed since the exchange proposal was approved (so that Governance has time to veto).
This check here:
https://github.com/celo-org/celo-proposals/pull/224/files#diff-fdeefbb76f60238f3814cb1238cb13ba0914ae709ec9856f368d49399900fa00R115